File: pset2/readability/readability.c
#include <cs50.h>
#include <ctype.h>
#include <math.h>
#include <stdio.h>
#include <string.h>

int main(void)
{
    string text = get_string("Text: ");

    int letters = 0;
    int words = 0;
    int sentences = 0;
    int length = strlen(text);

    if (length > 0)
    {
        words = 1; // Assume at least one word if the text is not empty
    }

    for (int i = 0; i < length; i++)
    {
        char c = text[i];
        if (isalpha(c))
        {
            letters++;
        }
        else if (c == ' ')
        {
            words++;
        }
        else if (c == '.' || c == '!' || c == '?')
        {
            sentences++;
        }
    }

    float L = 0;
    float S = 0;

    if (words > 0)
    {
        L = (float) letters / words * 100.0;
        S = (float) sentences / words * 100.0;
    }

    float index = 0.0588 * L - 0.296 * S - 15.8;
    int grade = round(index);

    if (grade < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (grade >= 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %i\n", grade);
    }
}

File: pset6/sentimental-readability/readability.py
import cs50
import math

text = cs50.get_string("Text: ")

letter_count = 0
word_count = 1  # Start with 1, assuming non-empty text
sentence_count = 0

for char in text:
    if char.isalpha():
        letter_count += 1
    elif char.isspace():
        word_count += 1
    elif char in ['.', '!', '?']:
        sentence_count += 1

# Adjust word count for texts starting/ending with space or having multiple spaces
# A more Pythonic way to count words:
words = text.split()
word_count = len(words)

# Handle potential division by zero if text is empty or has no words
if word_count == 0:
    L = 0.0
    S = 0.0
else:
    L = (letter_count / word_count) * 100
    S = (sentence_count / word_count) * 100

index = 0.0588 * L - 0.296 * S - 15.8
grade = round(index)

if grade >= 16:
    print("Grade 16+")
elif grade < 1:
    print("Before Grade 1")
else:
    print(f"Grade {grade}")
